code/modules/core_implant/cruciform/rituals/construction.dm

```diff
@@ -0,0 +1,246 @@
+	var/list/turf_contents = target.contents
+	
+	for(var/item_type in blueprint.materials)
+		var/located_raw = locate(item_type) in turf_contents
+		//single item check is handled there, rest of func is for stacked items or items with containers
+		if(!located_raw)
+			return FALSE
+
+		var/required_amount = blueprint.materials[item_type]
+        // I hope it is fast enough
+        // could have initialized it in glob
+		if(item_type in typesof(/obj/item/stack/))
+			var/obj/item/stack/stacked = located_raw
+			if(stacked.amount < required_amount)
+				return FALSE
+	return TRUE
+/datum/nt_blueprint/
+	var/name = "Report me"
+	var/build_path
+	var/list/materials
+	
+/datum/nt_blueprint/canister
+	name = "Biomatter Canister"
+	build_path = /obj/structure/reagent_dispensers/biomatter
+	materials = list(
+		/obj/item/stack/material/steel = 8,
+		/obj/item/stack/material/plastic = 2
+	)
+/datum/nt_blueprint/canister/large  
+	name = "Large Biomatter Canister"
+	build_path = /obj/structure/reagent_dispensers/biomatter/large
+	materials = list(
+		/obj/item/stack/material/steel = 16,
+		/obj/item/stack/material/plastic = 4,
+		/obj/item/stack/material/plasteel = 2,
+	)
+
+/datum/nt_blueprint/machinery
+
+/datum/nt_blueprint/machinery/obelisk
+	name = "Obelisk"
+	build_path = /obj/machinery/power/nt_obelisk
+	materials = list(
+		/obj/item/stack/material/plasteel = 10,
+		/obj/item/stack/material/gold = 5,
+		/CRUCIFORM_TYPE = 1
+	)
+/datum/nt_blueprint/machinery/bioprinter
+	name = "Biomatter Printer"
+	build_path = /obj/machinery/autolathe/bioprinter
+	materials = list(
+		/obj/item/stack/material/steel = 10,
+		/obj/item/stack/material/glass = 2,
+		/obj/item/stack/material/silver = 6,
+		/obj/item/weapon/storage/toolbox = 1
+	)
+
+/datum/nt_blueprint/machinery/solidifier
+	name = "Biomatter Solidifier"
+	build_path = /obj/machinery/biomatter_solidifier
+	materials = list(
+		/obj/item/stack/material/steel = 20,
+		/obj/item/stack/material/silver = 5,
+		/obj/structure/reagent_dispensers/biomatter = 1
+	)
+
+//cloner
+/datum/nt_blueprint/machinery/cloner
+	name = "Cloner Pod"
+	build_path = /obj/machinery/neotheology/cloner
+	materials = list(
+		/obj/item/stack/material/glass = 15,
+		/obj/item/stack/material/plasteel = 10,
+		/obj/item/stack/material/gold = 5,
+		/obj/item/stack/material/glass/plasmarglass = 10,
+	)
+
+/datum/nt_blueprint/machinery/reader
+	name = "Cruciform Reader"
+	build_path = /obj/machinery/neotheology/reader
+	materials = list(
+		/obj/item/stack/material/steel = 10,
+		/obj/item/stack/material/plasteel = 5,
+		/obj/item/stack/material/silver = 10,
+		/CRUCIFORM_TYPE = 1
+	)
+
+/datum/nt_blueprint/machinery/biocan
+	name = "Biomass tank"
+	build_path = /obj/machinery/neotheology/biomass_container
+	materials = list(
+		/obj/item/stack/material/gold = 5,
+		/obj/item/stack/material/plasteel = 5,
+		/obj/structure/reagent_dispensers/biomatter/large = 1
```

Comment:
Same with this one. Don't think these are craft-able.

Reply:
`/datum/nt_blueprint/canister` and `/datum/nt_blueprint/canister/large`